macro/vignettes/fluxmovement.Rmd

```diff
@@ -0,0 +1,146 @@
+
+The main restriction that will help us write the master equations is that as the time step goes to zero ($\delta t\rightarrow 0$), the probability of two or more events occuring simultaneously goes to zero. Put more succinctly, this just means two things can't happen at the same time.
+
+$$
+\frac{d}{dt}P(S_{1}) = -(f_{12} + f_{21})P(S_{1}) + 2f_{12}P(S_{2}) + 2f_{21}P(S_{3}) \\
+\frac{d}{dt}P(S_{2}) = -2f_{12}P(S_{2}) + f_{21}P(S_{1}) \\
+\frac{d}{dt}P(S_{3}) = -2f_{21}P(S_{3}) + f_{12}P(S_{1})
+$$
+
+We can look at a state transition diagram to see what the master equations tell us about the probability flow between states. The reason that states $S_{2}$ and $S_{3}$ send 2 times the rate into state $S_{1}$ is that because the 3 state master equation ignores identity of the individuals, state $S_{1}$ actually consists of two aggregated states, where person 1 is at patch 2 and person 2 is at patch 1, or where person 1 is at patch 1 and person 2 is at patch 2.
+
+```{r, echo=FALSE, fig.cap="State transition diagram", out.width = "60%", fig.align='center'}
+knitr::include_graphics("flux.png")
+```
+
+To solve for the stationary probability distribution, we set the ODEs equal to zero, and solve with the additional constraint that $P(S_{1}) + P(S_{2}) + P(S_{3}) = 1$. We get the solution:
+
+$$
+P(S_{1}) = \frac{2f_{12}f_{21}}{(f_{12} + f_{21})^{2}} \\
+P(S_{2}) = \frac{f_{21}^{2}}{(f_{12} + f_{21})^{2}} \\
+P(S_{3}) = \frac{f_{12}^{2}}{(f_{12} + f_{21})^{2}}
+$$
+
+## Stochastic Simulation
+
+Now we can use the functions provided by `marco` to set up a simple flux model. The only required paramter is the `rate_matrix` object, giving the $\{f_{ij}\}$ values.
+
+
+```{r}
+parameters <- new.env()
+parameters$rate_matrix <- matrix(c(0,1/6,1/5,0),byrow = TRUE,nrow = 2,ncol = 2)
+parameters$npatch <- 2
+
+parameters$location <- 1:2
+```
+
+Now we can set up and run the model. We'll use the function `flux_module` to make the simulation object, and then `mash_step` to simulate from it. We will run for 20,000 days, as we want a trajectory long enough to meaningfully compare to the stationary distribution we worked out above (it should reach the stationary distribution).
+
+```{r}
+simulation <- flux_module(parameters)
+simulation <- mash_step(simulation, 2e4, NULL)
+```
+
+After simulating the model, we are going to want to compare the simulated trajectory to the exact stationary probability vector we worked out earlier. To do that we'll write some code that takes the output trajectory of events and turns it into cumulative time spent in each state, to compare to the stationary probability distribution.
+
+```{r}
+trajectory <- Filter(Negate(is.null),simulation$trajectory)
```

Comment:
Are these dplyr functions? Filter and Negate?

Reply:
Nope. Base R.